ownCloud/Settings/UploadsSettingsSection.swift

```diff
@@ -0,0 +1,234 @@
+
+        photosRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+            if let photosSwitch = sender as? UISwitch {
+                self.isPhotoUploadEnabled = photosSwitch.isOn
+            }
+        }, title: "Photos".localized, value: isPhotoUploadEnabled)
+
+        photosWifiOnlyRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+            if let photosWifiOnlySwitch = sender as? UISwitch {
+                self.isPhotoWifiOnlyUploadsEnabled = photosWifiOnlySwitch.isOn
+            }
+        }, title: "Upload pictures via WiFi only".localized, value: isPhotoWifiOnlyUploadsEnabled)
+
+        photosSelectedPathRow = StaticTableViewRow(subtitleRowWithAction: { (_, _) in
+            // TODO: Use a more advanced version of ClientQueryViewController to select the path
+        }, title: "Photo upload path".localized, subtitle: photoSelectedPath, accessoryType: .disclosureIndicator)
+    }
+
+    private func createVideoRows() {
+
+        videosRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+            if let videosSwitch = sender as? UISwitch {
+                self.isVideoUploadEnabled = videosSwitch.isOn
+            }
+        }, title: "Videos".localized, value: isVideoUploadEnabled)
+
+        videosWifiOnlyRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+            if let videosWifiOnlySwitch = sender as? UISwitch {
+                self.isVideoWifiOnlyUploadsEnabled = videosWifiOnlySwitch.isOn
+            }
+        }, title: "Upload videos via WiFi only".localized, value: isVideoWifiOnlyUploadsEnabled)
+
+        videosSelectedPathRow = StaticTableViewRow(subtitleRowWithAction: { (_, _) in
+            // TODO: Use a more advanced version of ClientQueryViewController to select the path
+        }, title: "Video upload path".localized, subtitle: videoSelectedPath, accessoryType: .disclosureIndicator)
+    }
+
+    private func createCommonRows() {
+        backgroundUploadsRow = StaticTableViewRow(switchWithAction: { (_, sender) in
+            if let backgroundsSwitch = sender as? UISwitch {
+                self.backgroundUploadsEnabled = backgroundsSwitch.isOn
+            }
+        }, title: "Background uploads".localized, value: backgroundUploadsEnabled)
+    }
+
+    /// Create and configure all the rows the Uploads Section has in the settings view.
+    func updateUI() {
+
+        if isPhotoUploadEnabled {
+
+            let photoUploadIndex = rows.index(of: photosRow!)!
+
+            if !rows.contains(photosWifiOnlyRow!) {
+
+                insert(row: photosWifiOnlyRow!, at: photoUploadIndex + 1, animated: true)
+                insert(row: photosSelectedPathRow!, at: photoUploadIndex + 2, animated: true)
+            }
+
+        } else {
+            isPhotoWifiOnlyUploadsEnabled = false
+            remove(rows: [photosWifiOnlyRow!, photosSelectedPathRow!], animated: true)
+            userDefaults.removeObject(forKey: UploadsPhotoSelectedPathKey)
```

Comment:
What I'm going to do here is to remove the line:
```Swift
userDefaults.removeObject(forKey: UploadsPhotoSelectedPathKey)
```

Should be enough setting the property ```isPhotoWifiOnlyUploadsEnabled``` to false instead of deleting it.